R/utils.R

```diff
@@ -393,8 +394,16 @@
             "'. Consider using the 'name' argument to specify alternative ",
             "names.", call. = FALSE)
     }
-    reducedDim(x, name) <- values
-    return(x)
+    if( !identical(rownames(as.matrix(values)), 
+                  colnames(assay(x, assay.type))) ){
+        stop("Samples of the dissimilarity matrix should be the same as the",
+                " samples in columns of the assay specified with 'assay.type'.",
+                " The result is not added to reducedDim.")
```

Comment:
This checking should be prior to the if sentence above. User should get only the error (warning is not related to this issue).

```
> addDissimilarity(tse, method = "overlap", detection = 0.25, transposed = TRUE)
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by ‘tidytree’
Found more than one class "phylo" in cache; using the first, from namespace 'phyloseq'
Also defined by ‘tidytree’
Error in .add_values_to_reducedDims(x, values = as.matrix(res), name = name,  : 
  Samples of the dissimilarity matrix should be the same as the samples in columns of the assay specified with 'assay.type'. The result is not added to reducedDim.
In addition: Warning message:
The following values are already present in `reducedDims` and will be overwritten: 'overlap'. Consider using the 'name' argument to specify alternative names
```

Reply:
I think that `colnames(assay(x, assay.type))` is equal to `colnames(x)`? Better to use the shorter form?